clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3813,6 +4489,12 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return true;
   }
 
+  bool handleNonDecompStruct(const CXXRecordDecl *, ParmVarDecl *PD,
+                             QualType ParamTy) final {
+    addParam(PD, ParamTy, SYCLIntegrationHeader::kind_std_layout);
```

Comment:
Please replace with //TODO and unsupportedFreeFunctionParamType()

Reply:
Done.